chipsec/utilcmd/lock_check_cmd.py

```diff
@@ -0,0 +1,159 @@
+from argparse import ArgumentParser
+
+from chipsec.command import BaseCommand
+from chipsec.hal.locks import locks, LockResult
+from chipsec.defines import is_set
+from chipsec.chipset import CONSISTENCY_CHECKING
+
+
+class LOCKCHECKCommand(BaseCommand):
+    """
+    >>> chipsec_util check list
+    >>> chipsec_util check lock <lockname>
+    >>> chipsec_util check all
+
+    Examples:
+
+    >>> chipsec_util check list
+    >>> chipsec_util check lock DebugLock
+    >>> chipsec_util check all
+    """
+
+    version = "0.5"
+
+    def requires_driver(self):
+        parser = ArgumentParser(prog='chipsec_util check', usage=LOCKCHECKCommand.__doc__)
+
+        parser_lockname = ArgumentParser(add_help=False)
+        parser_lockname.add_argument('lockname', type=str, nargs='+', help="locknames")
+
+        subparsers = parser.add_subparsers()
+
+        # list
+        parser_list = subparsers.add_parser('list')
+        parser_list.set_defaults(func=self.list_locks)
+
+        # checkall
+        parser_checkall = subparsers.add_parser('all')
+        parser_checkall.set_defaults(func=self.checkall_locks)
+
+        # check
+        parser_check = subparsers.add_parser('lock', parents=[parser_lockname])
+        parser_check.set_defaults(func=self.check_lock)
+
+        parser.parse_args(self.argv[2:], namespace=self)
+
+        return True
+
+    def log_key(self):
+        self.logger.log("""
+KEY:
+\tLock Name - Name of Lock within configuration file
+\tState - Lock Configuration
+\t\tUndefined - Lock is not defined within configuration
+\t\tUndoc -  Lock is missing configuration information
+\t\tHidden - Lock is in a disabled or hidden state (unable to read the lock)
+\t\tUnlocked - Lock does not match value within configuration
+\t\tLocked - Lock matches value within configuration
+\t\tRW/O - Lock is identified as register is RW/O""")
+
+    def log_header(self):
+        ret = '{:^27}|{:^16}|{:^16}\n{}'.format('Lock Name', 'State', 'Consistent', '-' * 58)
+        if not self.logger.HAL:
+            self.logger.log(ret)
+        return "\n\n{}".format(ret)
+
+    def list_locks(self):
+        self.logger.log('Locks identified within the configuration:')
+        for lock in self._locks.get_locks():
+            self.logger.log(lock)
+        self.logger.log('')
+        return
+
+    def checkall_locks(self):
+        locks = self._locks.get_locks()
+        if not locks:
+            self.logger.log('Did not find any locks')
+            return
+        res = self.log_header()
+        if self.logger.VERBOSE:
+            self.log_key()
+        for lock in locks:
+            is_locked = self._locks.is_locked(lock)
+            res = "{}\n{}".format(res, self.check_log(lock, is_locked))
+        if self.logger.HAL:
+            self.logger.log(res)
+        return
+
+    def check_lock(self):
+        res = self.log_header()
+        if self.logger.VERBOSE:
+            self.log_key()
+        for lock in self.lockname:

```

Comment:
You don't have it documented anywhere that you allow multiple locks to be inputted in a single command line.